btree/btree.c

```diff
@@ -735,35 +770,97 @@ enum {
 /** Direction of move in node_move(). */
 enum dir {
 	/** Move (from right to) left. */
-	D_LEFT,
+	D_LEFT = 1,
 	/** Move (from left to) right. */
 	D_RIGHT
 };
 
 struct nd;
 struct slot;
+
+/**
+ *  Different types of btree node formats are supported. While the basic btree
+ *  operations remain the same, the differences are encapsulated in the nodes
+ *  contained in the btree.
+ *  Each supported node type provides the same interface to implement the btree
+ *  operations so that the node-specific changes are captured in the node
+ *  implementation.
+ */
 struct node_type {
 	uint32_t                   nt_id;
 	const char                *nt_name;
 	const struct m0_format_tag nt_tag;
+
+	/** Initializes newly allocated node */
+	void (*nt_init)(const struct nd *node, int shift, int ksize, int vsize);
+
+	/** Cleanup of the node if any before deallocation */
+	void (*nt_fini)(const struct nd *node);
+
+	/** Returns count of records in the node */
 	int  (*nt_count)(const struct nd *node);
+
+	/** Returns the space (in bytes) available in the node */
 	int  (*nt_space)(const struct nd *node);
+
+	/** Returns level of this node in the btree */
 	int  (*nt_level)(const struct nd *node);
+
+	/** Returns size of the node (as a shift value) */
 	int  (*nt_shift)(const struct nd *node);
+
+	/** Returns unique FID for this node */
 	void (*nt_fid)  (const struct nd *node, struct m0_fid *fid);
+
+	/** Returns record (KV pair) for specific index. */
 	void (*nt_rec)  (struct slot *slot);
+
+	/** Returns Key at a specifix index */
 	void (*nt_key)  (struct slot *slot);
+
+	/** Returns Child pointer (in segment) at specific index */
 	void (*nt_child)(struct slot *slot, struct segaddr *addr);
+
+	/**
+	 *  Returns TRUE if node has space to fit a new entry whose key and
+	 *  value length is provided in slot.
+	 */
 	bool (*nt_isfit)(struct slot *slot);
+
+	/**
+	 *  Node changes related to last record have completed; any post
+	 *  processing related to the record needs to be done in this function.
+	 */
 	void (*nt_done) (struct slot *slot, struct m0_be_tx *tx, bool modified);
+
+	/** Make space in the node for inserting new entry at specific index */
```

Comment:
Please keep the grammatical structure consistent: 3rd person singular present tense indicative mood active voice: "makes", "changes", etc.